3700bridge

```diff
@@ -84,24 +84,46 @@ class Bridge(object):
 
         self.bridge_id = deserialize_address(bridge_id)
         self.ports = [Port(i, lan) for i, lan in enumerate(lans)]
-        self.update_root(self.bridge_id, None, 0)
-        self.routing_table = dict()
+
+        self.root_bridge = bridge_id
+        self.root_port = None
+        self.root_cost = 0
+        self.next_bridge = None
+        self.root_bpdu = Packet.bpdu(self.bridge_id, self.bridge_id, 0)
+
+        self.reset_routing_table()
 
     def reset_routing_table(self):
         self.routing_table = dict()
 
-    def update_root(self, root_bridge: str, root_port: int, root_cost: int):
-        self.root_bridge = root_bridge
-        self.reset_routing_table()
-        print('New root: {}/{}'.format(serialize_address(self.bridge_id), root_bridge))
+    def update_root(self, low_bpdu: Packet):
+        changed = False
+
+        new_root = deserialize_address(low_bpdu.message['root'])
+        new_port = deserialize_address(low_bpdu.port)
+        new_cost = low_bpdu.message['cost'] + 1
+        new_next = deserialize_address(low_bpdu.message['id'])
+
+        if self.root_bridge != new_root:
+            self.root_bridge = new_root
+            changed = True
+            print('New root: {}/{}'.format(serialize_address(self.bridge_id), new_root))
 
-        self.update_root_port(root_port, root_cost)
+        if self.root_port != new_port:
+            self.update_root_port(new_port)
+            changed = True
 
-    def update_root_port(self, root_port: int, root_cost: int):
+        if changed:
+            self.root_cost = new_cost
+            self.next_bridge = new_next
+            self.root_bpdu = low_bpdu
+
+    def update_root_port(self, root_port: int):
         self.root_port = root_port
-        self.root_cost = root_cost
         print('Root port: {}/{}'.format(serialize_address(self.bridge_id), root_port))
 
+        self.reset_routing_table()
```

Comment:
you should only reset if any of those changed. if nothing changes we shouldn't reset the routing table

Reply:
ah wait. misread this.

Reply:
although i think we should be flushing the routing table here: https://github.com/aem/cs3700-project2/pull/3/files/8c2fdd6465d69fbe424ff3d5c95f93f98f6f5495#diff-e9f2bd50fd95eef919abd747ef089a7aR116